CodeEdit/Features/LSP/Registry/InstallationQueueManager.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  InstallationQueueManager.swift
+//  CodeEdit
+//
+//  Created by Abe Malla on 3/13/25.
+//
+
+import Foundation
+
+/// A class to manage queued installations of language servers
+final class InstallationQueueManager {
+    static let shared: InstallationQueueManager = .init()
+
+    /// The maximum number of concurrent installations allowed
+    private let maxConcurrentInstallations: Int = 2
+    /// Queue of pending installations
+    private var installationQueue: [(RegistryItem, (Result<Void, Error>) -> Void)] = []
+    /// Currently running installations
+    private var runningInstallations: Set<String> = []
+    /// Installation status dictionary
+    private var installationStatus: [String: PackageInstallationStatus] = [:]
+
+    /// Add a package to the installation queue
+    func queueInstallation(package: RegistryItem, completion: @escaping (Result<Void, Error>) -> Void) {
+        // If we're already at max capacity and this isn't already running, mark as queued
+        if runningInstallations.count >= maxConcurrentInstallations && !runningInstallations.contains(package.name) {
+            installationStatus[package.name] = .queued
+            installationQueue.append((package, completion))
+
+            // Notify UI that package is queued
+            DispatchQueue.main.async {
```

Comment:
We might consider using the activity viewer here to show the user something is going on.